pandas-stubs/core/series.pyi

```diff
@@ -1025,9 +1032,99 @@ class Series(IndexOpsMixin, NDFrame, Generic[S1]):
         axis: SeriesAxisType | None = ...,
         ignore_index: _bool = ...,
     ) -> Series[S1]: ...
+    @overload
+    def astype(
+        self,
+        dtype: Literal["int", "int32"]
+        | type[int]
+        | type[bool]
+        | pd.Int8Dtype
+        | pd.Int16Dtype
+        | pd.Int32Dtype
+        | pd.Int64Dtype
+        | type[np.int8]
+        | type[np.int16]
+        | type[np.int32]
+        | type[np.int64]
+        | type[np.uint8]
+        | type[np.uint16]
+        | type[np.uint32]
+        | type[np.uint64]
+        | type[np.intp]
+        | type[np.uintp]
+        | type[np.integer]
+        | type[np.byte]
+        | type[np.ubyte],
+        copy: _bool = ...,
+        errors: IgnoreRaise = ...,
+    ) -> Series[int]: ...
+    @overload
+    def astype(
+        self,
+        dtype: type[_str] | pd.StringDtype | Literal["str"],
+        copy: _bool = ...,
+        errors: IgnoreRaise = ...,
+    ) -> Series[_str]: ...
+    @overload
+    def astype(
+        self,
+        dtype: bytes,
+        copy: _bool = ...,
+        errors: IgnoreRaise = ...,
+    ) -> Series[bytes]: ...
+    @overload
+    def astype(
+        self,
+        dtype: pd.Float32Dtype
+        | pd.Float64Dtype
+        | type[np.float16]
+        | type[np.float32]
+        | type[np.float64]
+        | type[np.float96]
+        | type[np.float128]
+        | type[np.floating]
+        | type[float]
+        | Literal["float"],
+        copy: _bool = ...,
+        errors: IgnoreRaise = ...,
+    ) -> Series[float]: ...
+    @overload
+    def astype(
+        self,
+        dtype: type[np.complex64]
+        | type[np.complex128]
+        | type[np.complex192]
+        | type[np.complex256]
+        | type[complex]
+        | Literal["complex"],
+        copy: _bool = ...,
+        errors: IgnoreRaise = ...,
+    ) -> Series[complex]: ...
+    @overload
+    def astype(
+        self,
+        dtype: pd.BooleanDtype | Literal["bool"],
```

Comment:
I'm a little worried about removing `type[bool]` here, but I think  you need to do the same thing as you did with `_str`, i.e., try adding `type[_bool]`

Reply:
i'll try